### DISCUSSION TITLE
Facing error while updating user Identity

### DISCUSSION THREAD

**Discussion by jimit-raval-roambee** [ory/kratos#3072] [Q&A]
I'm using **"@ory/client": "^1.1.7"** to update and create Identity.

Below is my code :

`import { Configuration, IdentityApi, FrontendApi, Session } from "@ory/client";
private readonly admin: IdentityApi;
this.admin = new IdentityApi(
	new Configuration({
		basePath: config.oryAdminUrl,
		baseOptions: { withCredentials: true }
	})
);

// First we are fetching all the existing Identity
// here we are checking if Identity is already exists then update it.

await this.admin.updateIdentity({
  id: exists.id,
  updateIdentityBody: {
	  schema_id: exists.schema_id,
	  state: exists.state || "inactive",
	  traits: {
		  ...exists.traits,
		  name: { first: user.first_name.trim(), last: user.last_name.trim() },
		  email: email
	  },
	  credentials: user.pw ? { password: { config: { hashed_password: user.pw } } } : undefined
  }
});`

Getting below error while updating the Identity of existing user Identity.

`{"args":[{"message":"Request failed with status code 500","name":"Error","stack":"Error: Request failed with status code 500\n    at createError (/node_modules/@ory/client/node_modules/axios/lib/core/createError.js:16:15)\n    at settle (/node_modules/@ory/client/node_modules/axios/lib/core/settle.js:17:12)\n    at IncomingMessage.handleStreamEnd (/node_modules/@ory/client/node_modules/axios/lib/adapters/http.js:269:11)\n    at IncomingMessage.emit (node:events:549:35)\n    at endReadableNT (node:internal/streams/readable:1359:12)\n    at processTicksAndRejections (node:internal/process/task_queues:82:21)","config":{"url":"[http://localhost:4434/admin/identities/864d0ae6-90ba-4b3f-a082-0164d816e67f","method":"put","data":"{\"schema_id\":\"default\",\"state\":\"active\",\"traits\":{\"name\":{\"first\":\"John\",\"last\":\"Doe\"},\"email\":\"john.doe@gmail.com\"},\"credentials\":{\"password\":{\"config\":{\"hashed_password\":\"$fh843fy4738fh4837h487gfh478hf473fh4837\"}}}}","headers":{"Accept":"application/json](http://localhost:4434/admin/identities/864d0ae6-90ba-4b3f-a082-0164d816e67f%22,%22method%22:%22put%22,%22data%22:%22%7B/%22schema_id/%22:/%22default/%22,/%22state/%22:/%22active/%22,/%22traits/%22:%7B/%22name/%22:%7B/%22first/%22:/%22John/%22,/%22last/%22:/%22Doe/%22%7D,/%22email/%22:/%22john.doe@gmail.com/%22%7D,/%22credentials/%22:%7B/%22password/%22:%7B/%22config/%22:%7B/%22hashed_password/%22:/%22$fh843fy4738fh4837h487gfh478hf473fh4837/%22%7D%7D%7D%7D%22,%22headers%22:%7B%22Accept%22:%22application/json), text/plain, */*","Content-Type":"application/json","User-Agent":"axios/0.21.4","Content-Length":236},"transformRequest":[null],"transformResponse":[null],"timeout":0,"withCredentials":true,"xsrfCookieName":"XSRF-TOKEN","xsrfHeaderName":"X-XSRF-TOKEN","maxContentLength":-1,"maxBodyLength":-1,"transitional":{"silentJSONParsing":true,"forcedJSONParsing":true,"clarifyTimeoutError":false}}},"OryService"]}`

I've followed this link : https://www.ory.sh/docs/guides/upgrade/sdk-v1
I'm not sure what I'm missing here.
Thanks in advance for help.

**Comment by aeneasr** [comment]
Hey, it looks like you're using Ory Kratos in self-hosting. The best option to understand what is going on is to check the Ory Kratos server logs :)

**Comment by vinckr** [comment]
Hello @jimit-raval-roambee 
You are using ory/client - the SDK for Ory Network.
For self-hosting Ory Kratos please use kratos-client. See the documentation [here](https://www.ory.sh/docs/kratos/sdk/overview).
Moving this to ory/kratos.

**Comment by Saifbenhmida** [comment]
use the ory/client +  IdentityApi ( updateIdentity )
